### PR TITLE
Fix nested attribute name and generated custom value method name conflicts

### PR DESCRIPTION
**Pull request by bendbennett**
Closes: #76

Refer to [issue comment](https://github.com/hashicorp/terraform-plugin-codegen-framework/issues/76#issuecomment-1781026098) for details regarding the changes in this PR.

### CODE REVIEW

**Comment by bflad** [comment]
For additional context: I do not believe we can we get away with unexported (lowercase first letter) field names which would wholly eliminate this problem since the framework's reflection needs them to be exported.

So given that, I don't think we have much choice in this situation -- we need to munge the field names in some fashion. Another option could be to potentially suffix (these or all) field names with "Attribute"/"Field" or similar. Not sure if that's any better or worse than prefixing in terms of developers trying to find field names that are now mismatched with the attribute names.

File: internal/schema/framework_identifier.go [hunk 1]
```diff
@@ -42,6 +42,26 @@ func (identifier FrameworkIdentifier) ToCamelCase() string {
 	return string(unicode.ToLower(firstLetter)) + pascal[size:]
 }
 
+// ToPrefixCamelCase will return a camel case formatted string of the identifier,
+// prefixed with a camel-cased version of the supplied name if the identifier is
+// a generated custom value method name.
+// Example:
+//   - equal(something) -> somethingEqual
+//   - type(something) -> somethingType
+func (identifier FrameworkIdentifier) ToPrefixCamelCase(name string) string {
```
Comment (austinvalle) [inline internal/schema/framework_identifier.go:51]: nit: Would changing the name of the input parameter to `prefix` make more sense in this context? Same for the method below

**Review by austinvalle — APPROVED** [review]
LGTM, one nit :shipit:

File: internal/schema/framework_identifier.go (hunk 1 above)
Comment (bendbennett) [reply internal/schema/framework_identifier.go]: Fair point. Have updated.

**Pull request merged**